helm-charts/splunk-otel-collector/templates/operator/_helpers.tpl

```diff
@@ -0,0 +1,64 @@
+{{- define "validation-rules" -}}
+{{- $tracesEnabled := or (include "splunk-otel-collector.platformTracesEnabled" .) (include "splunk-otel-collector.o11yTracesEnabled" .) -}}
+{{- $endpointOverridden := and .Values.operator.instrumentation.spec .Values.operator.instrumentation.spec.exporter .Values.operator.instrumentation.spec.exporter.endpoint (ne .Values.operator.instrumentation.spec.exporter.endpoint "") -}}
+{{- if and .Values.operator.enabled $tracesEnabled (not $endpointOverridden) -}}
+  {{/*  If no endpoint override was provided, the environment variable/tag must be set by the agent, gateway, or instrumentation.*/}}
+  {{- if or (not .Values.environment) (eq .Values.environment "") -}}
+    {{- $envSet := false -}}
+    {{- if and .Values.operator.instrumentation.spec .Values.operator.instrumentation.spec.env -}}
+      {{- range .Values.operator.instrumentation.spec.env -}}
+        {{- if and (eq .name "OTEL_RESOURCE_ATTRIBUTES") .value -}}
+          {{- $envSet = true -}}
```

Comment:
Having `OTEL_RESOURCE_ATTRIBUTES` doesn't mean `development.environment` is set, user can set any other resource attribute.

Reply:
True, done.